src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java21/ExpectedStringTemplateBasic.txt

```diff
@@ -563,23 +542,22 @@ COMPILATION_UNIT -> COMPILATION_UNIT [2:0]
         |       |           |--EMBEDDED_EXPRESSION -> EMBEDDED_EXPRESSION [64:26]
         |       |           |   `--IDENT -> x [64:26]
         |       |           |--EMBEDDED_EXPRESSION_END -> } [64:28]
-        |       |           |--EMBEDDED_EXPRESSION -> EMBEDDED_EXPRESSION [64:33]
-        |       |           |   `--IDENT -> y [64:33]
         |       |           |--STRING_TEMPLATE_CONTENT -> . [64:29]
         |       |           |--EMBEDDED_EXPRESSION_BEGIN -> \{ [64:30]
+        |       |           |--EMBEDDED_EXPRESSION -> EMBEDDED_EXPRESSION [64:33]
+        |       |           |   `--IDENT -> y [64:33]
         |       |           |--EMBEDDED_EXPRESSION_END -> } [64:35]
-        |       |           |--EMBEDDED_EXPRESSION -> EMBEDDED_EXPRESSION [64:40]
-        |       |           |   `--IDENT -> x [64:40]
         |       |           |--STRING_TEMPLATE_CONTENT -> . [64:36]
         |       |           |--EMBEDDED_EXPRESSION_BEGIN -> \{ [64:37]
+        |       |           |--EMBEDDED_EXPRESSION -> EMBEDDED_EXPRESSION [64:40]
+        |       |           |   `--IDENT -> x [64:40]
```

Comment:
As you can see, the old implementation caused us to have a bunch of out of order elements in the AST